src/ComputeJoinPoint/index.ts

```diff
@@ -0,0 +1,49 @@
+import { IMinAndMax } from './interface';
+
+export class ComputeJoinPoint {
+  private getMinAndMax(num1: number, num2: number): IMinAndMax {
+    if (num1 < num2)
+      return {
+        min: num1,
+        max: num2
+      };
+    return {
+      min: num2,
+      max: num1
+    };
+  }
+
+  private addition(num: number) {
+    return num
+      .toString()
+      .split('')
+      .reduce((acc: number, current: string) => {
+        return acc + parseInt(current);
+      }, num);
+  }
+
+  static get(num1: number, num2: number): number {
+    if (num1 === num2) return num1;
+    const compute = new ComputeJoinPoint();
+
+    const numbers = compute.getMinAndMax(num1, num2);
+
+    const limit = 20e6;
+
+    // start add the smaller summation until it is greater than bigger summation
+    let sumOfMin = numbers.min;
+    let sumOfMax = numbers.max;
+
+    while (sumOfMin < limit && sumOfMax < limit) {
+      if (sumOfMin > sumOfMax) {
+        while (sumOfMax < sumOfMin) {
+          sumOfMax = compute.addition(sumOfMax);
+          if (sumOfMax === sumOfMin) return sumOfMin;
+        }
+      }
+      sumOfMin = compute.addition(sumOfMin);
+      if (sumOfMax === sumOfMin) return sumOfMin;
+    }
+    throw Error(`${[num1, num2]} has no join point`);
+  }
```

Comment:
This shouldn't be a static function. Static should be very small and call some instance values and the name of the class shouldn't have any verbs, so it should be something like this
```
JoinPointFinder.compute(num1, num2) 
```

and the static function should be something like this
```
static compute(num1: number, num2: number): number {
  const jointPointFinder = new JoinPointFinder(num1, num2)
  return jointPointFinder.find()
}
```